repair/row_level.cc

```diff
@@ -2255,36 +2255,41 @@ static future<> repair_put_row_diff_with_rpc_stream_handler(
         uint32_t repair_meta_id,
         rpc::sink<repair_stream_cmd> sink,
         rpc::source<repair_row_on_wire_with_cmd> source) {
-    return do_with(false, repair_rows_on_wire(), [&repair, from, src_cpu_id, dst_cpu_id, repair_meta_id, sink, source] (bool& error, repair_rows_on_wire& current_rows) mutable {
-        return repeat([&repair, from, src_cpu_id, dst_cpu_id, repair_meta_id, sink, source, &current_rows, &error] () mutable {
-            return source().then([&repair, from, src_cpu_id, dst_cpu_id, repair_meta_id, sink, source, &current_rows, &error] (std::optional<std::tuple<repair_row_on_wire_with_cmd>> row_opt) mutable {
-                if (row_opt) {
+    std::exception_ptr outer_exception;
+    bool error = false;
```

Comment:
We could move the std::exception_ptr ep here and get rid of the error. But it is fine to keep both.

Reply:
I wanted the change to be as small as possible. We can have a follow-up cleanup round later.